### ISSUE TITLE
Support "Disk device, write temporary" mode for iso disc images

### ISSUE THREAD

**Issue by theraininsky** [ArsenalRecon/Arsenal-Image-Mounter#46]
It would be great if this could be achieved.

**Comment by LTRData** [comment]
For ISO images, this driver emulates CD/DVD drives which are always read-only. It is not possible to introduce write support by adding a write overaly like for hard disk images. 

CD burners for example, work in a different way by writing an image at once instead of modifying individual files.

However, UDF images are possible to modify much like a hard disk file systems and it is technically possible to have a hard disk partition formatted with UDF file system too and in such cases, it is possible to use write overlay features.

**Comment by LTRData** [comment]
For ISO9660, no, that is not possible. You need to copy the files on it somewhere else and make modifications there. The OS file system driver for ISO9660 does not support modifying files at all, even if the underlying storage device would support writing.

For UDF, it is possible to modify in some cases. Particularly if it is located on a hard disk partition. 

I think that what you are looking for is not a disk write filter but more a kind of file system write filter, where modifications to a file system could store changes somewhere else and where it is not limited by the file system driver in the OS. Something like Unified Write Filter that is available in enterprise editions of Windows etc.

**Comment by theraininsky** [comment]
Yes, all i want is to support write operations on mounted drive and store the diffs elsewhere without modify the image itself.
I've tried Unified Write Filter you mentioned but unfortuntelly it does not work on emulated CD/DVD drives no matter i choose UDF images or ISO9660 images.

**Comment by LTRData** [comment]
> Yes, all i want is to support write operations on mounted drive and store the diffs elsewhere without modify the image itself.
> I've tried Unified Write Filter you mentioned but unfortuntelly it does not work on emulated CD/DVD drives no matter i choose UDF images or ISO9660 images.

Yes unfortunately that is the case for that particular driver. But something like that should be possible to use as starting point for developing something like what you want, if it has not already been done somewhere.

**Comment by theraininsky** [comment]
> > Yes, all i want is to support write operations on mounted drive and store the diffs elsewhere without modify the image itself.
> > I've tried Unified Write Filter you mentioned but unfortuntelly it does not work on emulated CD/DVD drives no matter i choose UDF images or ISO9660 images.
> 
> Yes unfortunately that is the case for that particular driver. But something like that should be possible to use as starting point for developing something like what you want, if it has not already been done somewhere.

Yes, I think there might be another file system filter tool just like MS's Unified Write Filter in the near future, but more universal and versatile.
